app/models.py

```diff
@@ -391,6 +429,34 @@ def from_id(self, user_id):  # Might come in useful
         guid, auth_user_type = user_id.split(USER_ID_DELIMITER)
         return self.query.filter_by(guid=guid, auth_user_type=auth_user_type).one()
 
+    def is_agency_admin(self, ein=default_agency_ein):
+        """
+        Determine if a user is an admin for the specified agency.
+        :param ein: Agency EIN (4 Character String)
+        :return: Boolean
+        """
+        for agency in self.agency_users.all():
+            if agency.agency_ein == ein:
+                return agency.is_agency_admin
+        return False
+
+    def is_agency_active(self, ein=default_agency_ein):
+        """
+        Determine if a user is active for the specified agency.
+        :param ein: Agency EIN (4 Character String)
+        :return: Boolean
+        """
+        for agency in self.agency_users.all():
+            if agency.agency_ein == ein:
+                return agency.is_agency_active
+        return False
+
+    def agencies_for_forms(self):
+        agencies = self.agencies.query.with_entities(Agencies.ein, Agencies.name).all()
```

Comment:
How are you populating your database? I wasn't able to reproduce with the data that i have.